File: src/main/java/space/bielsolososdev/noto/domain/users/repository/RoleRepository.java
package space.bielsolososdev.noto.domain.users.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import space.bielsolososdev.noto.domain.users.model.Role;

import java.util.Optional;

public interface RoleRepository extends JpaRepository<Role, Long> {

    Optional<Role> findByName(String name);
}

File: src/main/kotlin/space/bielsolososdev/noto/domain/users/repository/RoleRepository.kt
package space.bielsolososdev.noto.domain.users.repository

import org.springframework.data.jpa.repository.JpaRepository
import space.bielsolososdev.noto.domain.users.model.Role
import java.util.Optional

interface RoleRepository : JpaRepository<Role, Long> {

    fun findByName(name: String): Optional<Role>
}

